test/next_ls/helpers/ast_helpers_variables_test.exs

```diff
@@ -0,0 +1,338 @@
+      assert length(refs) == 1
+      assert exists_in?(refs, {:delta, {74..74, 27..32}})
+
+      refs2 = Variables.list_variable_references(source, {74, 11})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:delta, {75..75, 18..23}})
+
+      refs3 = Variables.list_variable_references(source, {78, 7})
+      assert length(refs3) == 1
+      assert exists_in?(refs3, {:error, {78..78, 25..30}})
+    end
+
+    test "symbol shadowed by anonymus funciton", %{source: source} do
+      refs = Variables.list_variable_references(source, {84, 12})
+      assert length(refs) == 2
+      assert exists_in?(refs, {:alpha, {96..96, 7..12}})
+      assert exists_in?(refs, {:alpha, {100..100, 21..26}})
+    end
+
+    test "symbol defined in anonymus funciton", %{source: source} do
+      refs = Variables.list_variable_references(source, {86, 7})
+      assert length(refs) == 1
+      assert exists_in?(refs, {:charlie, {88..88, 15..22}})
+
+      refs2 = Variables.list_variable_references(source, {86, 22})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:delta, {88..88, 24..29}})
+
+      refs3 = Variables.list_variable_references(source, {86, 31})
+      assert length(refs3) == 1
+      assert exists_in?(refs3, {:alpha, {87..87, 20..25}})
+
+      refs4 = Variables.list_variable_references(source, {97, 22})
+      assert length(refs4) == 1
+      assert exists_in?(refs4, {:alpha, {97..97, 37..42}})
+
+      refs5 = Variables.list_variable_references(source, {98, 25})
+      assert length(refs5) == 1
+      assert exists_in?(refs5, {:alpha, {98..98, 51..56}})
+    end
+
+    test "symbols with 'receive' macro", %{source: source} do
+      refs = Variables.list_variable_references(source, {103, 12})
+      assert length(refs) == 2
+      assert exists_in?(refs, {:alpha, {110..110, 10..15}})
+      assert exists_in?(refs, {:alpha, {113..113, 17..22}})
+
+      refs2 = Variables.list_variable_references(source, {105, 19})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:bravo, {107..107, 17..22}})
+    end
+
+    test "symbols set with 'destructure'", %{source: source} do
+      refs = Variables.list_variable_references(source, {117, 18})
+      assert length(refs) == 1
+      assert exists_in?(refs, {:alpha, {118..118, 16..21}})
+    end
+
+    test "symbols set in macro", %{source: source} do
+      refs = Variables.list_variable_references(source, {121, 37})
+      assert length(refs) == 1
+      assert exists_in?(refs, {:variables, {122..122, 14..23}})
+
+      refs2 = Variables.list_variable_references(source, {124, 7})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:length, {127..127, 32..38}})
+
+      refs3 = Variables.list_variable_references(source, {123, 7})
+      assert length(refs3) == 1
+      assert exists_in?(refs3, {:var, {127..127, 17..20}})
+    end
+  end
+
+  defp exists_in?(results, var), do: Enum.find(results, &(&1 == var))
```

Comment:
I think this reimplementing Enum.member?/2

Reply:
Which also can be expressed as `assert item in results`

Reply:
yes, you are totally right. Sorry, I just had a little brain fart. Will fix.